presentation/common/src/presentation-common/EC.ts

```diff
@@ -171,29 +215,45 @@ export interface PropertyInfo {
   kindOfQuantity?: KindOfQuantityInfo;
   /** Extended type name of the ECProperty if it has one */
   extendedType?: string;
+  /** Navigation property info if the field is navigation type */
+  navigationPropertyInfo?: NavigationPropertyInfo;
 }
 
 /** @public */
 export namespace PropertyInfo {
   /** Serialize [[PropertyInfo]] to JSON */
   export function toJSON(info: PropertyInfo): PropertyInfoJSON {
-    return { ...info, classInfo: ClassInfo.toJSON(info.classInfo) };
+    if (!info.navigationPropertyInfo)
+      return { ...info, classInfo: ClassInfo.toJSON(info.classInfo) };
+
+    return { ...info, classInfo: ClassInfo.toJSON(info.classInfo), navigationPropertyInfo: NavigationPropertyInfo.toJSON(info.navigationPropertyInfo) };
   }
 
   /** Serialize [[PropertyInfo]] to compressed JSON */
   export function toCompressedJSON(propertyInfo: PropertyInfo, classesMap: { [id: string]: CompressedClassInfoJSON }): PropertyInfoJSON<string> {
     const { id, ...leftOverInfo } = propertyInfo.classInfo;
     classesMap[id] = leftOverInfo;
 
-    return {
+    const propertyInfoJSON: PropertyInfoJSON<string> = {
       ...propertyInfo,
       classInfo: propertyInfo.classInfo.id,
+      navigationPropertyInfo: propertyInfo.navigationPropertyInfo
+        ? NavigationPropertyInfo.toCompressedJSON(propertyInfo.navigationPropertyInfo, classesMap)
+        : undefined,
     };
+
+    if (!propertyInfoJSON.navigationPropertyInfo)
+      delete propertyInfoJSON.navigationPropertyInfo;
+
+    return propertyInfoJSON;
   }
 
   /** Deserialize [[PropertyInfo]] from JSON */
   export function fromJSON(json: PropertyInfoJSON): PropertyInfo {
-    return { ...json, classInfo: ClassInfo.fromJSON(json.classInfo) };
+    if (!json.navigationPropertyInfo)
+      return { ...json, classInfo: ClassInfo.fromJSON(json.classInfo) };
+
+    return { ...json, classInfo: ClassInfo.fromJSON(json.classInfo), navigationPropertyInfo: NavigationPropertyInfo.fromJSON(json.navigationPropertyInfo) };
```

Comment:
This approach works when only one attribute needs to be handled like this, but gets very complicated as soon as there's more than 1 such attribute. Instead, I suggest doing this:
```ts
return { 
  ...json, 
  classInfo: ClassInfo.fromJSON(json.classInfo), 
  ...(json.navigationPropertyInfo ? { navigationPropertyInfo: NavigationPropertyInfo.fromJSON(json.navigationPropertyInfo) } : undefined),
};
```

The same approach can be used in places where you add the attribute and then `delete` it if its value is `undefined`.